corehq/apps/app_manager/models.py

```diff
@@ -307,13 +317,13 @@ def get_action_properties(self, action):
         if 'external_id' in action_properties and action.external_id:
             yield 'external_id', action.external_id
         if 'update' in action_properties:
-            for name, path in action.update.items():
+            for name, path in list(action.update.items()):
```

Comment:
It's possible that it's not needed, but it's also possible that it is. The generator would fail if the thing that's iterating over this generator function mutates `action.update` during iteration.  Same below.